### PR TITLE
[capture] Batch mode for Ibex SCA

### PR DESCRIPTION
**Pull request by nasahlpa**
This commit adds the command handler for:
- ibex_sca_tl_write_batch_fvsr
- ibex_sca_tl_write_batch_fvsr_fix_address
- ibex_sca_tl_write_batch_random
- ibex_sca_tl_write_batch_random_fix_address
- ibex_sca_tl_write_fvsr
- ibex_sca_tl_write_random
- ibex_sca_tl_read_batch_fvsr
- ibex_sca_tl_read_batch_fvsr_fix_address
- ibex_sca_tl_read_batch_random
- ibex_sca_tl_read_batch_random_fix_address
- ibex_sca_tl_read_fvsr
- ibex_sca_tl_read_random
- ibex_sca_register_file_write_batch_fvsr
- ibex_sca_register_file_write_batch_random
- ibex_sca_register_file_write_fvsr
- ibex_sca_register_file_write_random
- ibex_sca_register_file_read_batch_fvsr
- ibex_sca_register_file_read_batch_random
- ibex_sca_register_file_read_fvsr

The device code is located in lowRISC/opentitan#22272

### CODE REVIEW

File: capture/capture_ibex.py [hunk 1]
```diff
@@ -143,38 +150,51 @@ def establish_communication(target, capture_cfg: CaptureConfig):
...
-        data = []
-        for i in range(0, 8):
-            data.append(random.randint(0, 65535))
-    return data
-
-
-def capture(scope: Scope, ot_ibex: OTIbex, capture_cfg: CaptureConfig,
-            project: SCAProject, target: Target):
+    data = []
+    data_fixed = 0xABBABABE
```
Comment (vogelpi) [inline capture/capture_ibex.py:178]: It's ABBA, BABE! :-D I like this one.

File: capture/configs/ibex_sca_chip.yaml [hunk 2]
```diff
@@ -7,25 +7,46 @@ target:
   output_len_bytes: 16
   protocol: "ujson"
   port: "/dev/ttyUSB1"
+  usb_serial: "205F355C3236"
```
Comment (vogelpi) [inline capture/configs/ibex_sca_chip.yaml:10]: Is this intentional? I guess it will not work in other people's setups right?

File: capture/configs/ibex_sca_cw310.yaml [hunk 3]
```diff
@@ -22,7 +23,26 @@ capture:
...
+  # which_test: ibex_sca_register_file_write_batch_random
+  # which_test: ibex_sca_register_file_write_fvsr
+  # which_test: ibex_sca_register_file_write_random
+  # which_test: ibex_sca_register_file_read_batch_fvsr
+  # which_test: ibex_sca_register_file_read_batch_random
+  # which_test: ibex_sca_register_file_read_fvsr
+  which_test: ibex_sca_register_file_read_random
+  # seed for PRNG to generate sequence of plaintexts and keys; Python random
+  # class on host, Mersenne twister implementation on OT SW.
+  batch_prng_seed: 0
```
Comment (vogelpi) [inline capture/configs/ibex_sca_cw310.yaml:48]: Missing newline

File: fault_injection/configs/pen.global_fi.ibex.char.sram_static.cw310.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+target:
```
Comment (vogelpi) [inline fault_injection/configs/pen.global_fi.ibex.char.sram_static.cw310.yaml:1]: This file seems unrelated to this PR

File: target/communication/fi_ibex_commands.py [hunk 5]
```diff
@@ -90,6 +90,15 @@ def ibex_char_sram_write(self) -> None:
...
 
+    def ibex_char_sram_static(self) -> None:
+        """ Starts the ibex.char.sram_static test.
+        """
+        # IbexFi command.
+        self._ujson_ibex_fi_cmd()
+        # CharSramWrite command.
+        time.sleep(0.01)
+        self.target.write(json.dumps("CharSramStatic").encode("ascii"))
+
```
Comment (vogelpi) [inline target/communication/fi_ibex_commands.py:101]: Is this change related to this PR?

**Review by vogelpi — APPROVED** [review]
LGTM, but there seems to be some changes in this PR which are not related to SCA but rather FI. This is probably accidental.

General question: Since the SRAM is scrambled with an ephemeral key, the leakage observed from the SRAM itself will change after every reset / scrambling key change. I guess people are aware and since we want to measure the leakage on the bus and inside Ibex this is probably fine. However, if more deterministic leakage would be desired, some unscrambled memory / device addresses should be used.

File: capture/configs/ibex_sca_chip.yaml (hunk 2 above)
Comment (nasahlpa) [reply capture/configs/ibex_sca_chip.yaml:10]: Fixed, thanks.

File: capture/configs/ibex_sca_cw310.yaml (hunk 3 above)
Comment (nasahlpa) [reply capture/configs/ibex_sca_cw310.yaml:48]: Fixed, thanks!

**Comment by nasahlpa** [comment]
> LGTM, but there seems to be some changes in this PR which are not related to SCA but rather FI. This is probably accidental.
> 
> General question: Since the SRAM is scrambled with an ephemeral key, the leakage observed from the SRAM itself will change after every reset / scrambling key change. I guess people are aware and since we want to measure the leakage on the bus and inside Ibex this is probably fine. However, if more deterministic leakage would be desired, some unscrambled memory / device addresses should be used.

Thanks for the review! Yes, the intention behind these tests is to analyze the bus and not the SRAM. 

I'll merge this PR after the device [PR](https://github.com/lowRISC/opentitan/pull/22272) is merged.
